client/src/pages/Details.js

```diff
@@ -1,26 +1,51 @@
 import styled from 'styled-components';
 import { useState } from 'react';
 import { v4 as uuid4 } from 'uuid';
-import { isValidEntry } from '../lib/validateFunctions';
 
 import EntryCard from '../components/EntryCard';
+import { isValidEntry } from '../lib/validateFunctions';
+import NoteTags from '../components/NoteTags';
+import { StarIconFilled } from '../icons/StarIconFilled';
 import { UnfoldIcon } from '../icons/UnfoldIcon';
 
 export default function Details({ member, updateMember, members }) {
   const initialEntry = {
     date: '',
     title: '',
-    remember: '',
+    remember: [],
   };
 
   const [newMember, setNewMember] = useState(
     members.find((newMember) => newMember.id === member.id)
   );
-
   const [entry, setEntry] = useState(initialEntry);
   const [entries, setEntries] = useState(newMember.entries ?? []);
   const [isError, setIsError] = useState(false);
   const [isUnfolded, setIsUnfolded] = useState(false);
+  const [tags, setTags] = useState([]);
+  const [savedNotes, setSavedNotes] = useState(findSavedNotes() ?? []);
```

Comment:
I think you can simplify this a little, you don't need all these states for derived values, simply use a const that will get reevaluated every time the component rerenders.

Think about `newMember`: it is derived from `member`, when you call `updateMember` the state higher up changes, the `members` prop changes in turn and the component rerenders - no need for an extra state.

I think you don't need `member` as a prop anyway, you only need the id, which will probably not change and thus prevent further unnecessary rerendering.

Reply:
I had to take a deeper look first, but you are right, rerendering of newMember is not needed! Thank you :-)